### PR TITLE
Spellchecking with codespell

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds `codespell` for checking word spellings
* Configures `codespell` to ignore several French words and all current translations configurations.
* Adds a `pre-commit` hook for performing these checks on commit

### Does this PR introduce a breaking change?

Yes, `codespell` is now a development dependency.

### Other information:

https://github.com/codespell-project/codespell

### CODE REVIEW

File: docs/notebooks/ensembles.ipynb [hunk 1]
```diff
@@ -17,8 +17,6 @@
     "\n",
     "from __future__ import annotations\n",
     "\n",
-    "from pathlib import Path\n",
-    "\n",
```
Comment (coxipi) [inline docs/notebooks/ensembles.ipynb:21]: This was just unused? This is unrelated to the general idea of the PR correct?

File: CHANGES.rst [hunk 2]
```diff
@@ -24,12 +24,14 @@ Bug fixes
...
+* Several spelling mistakes have been corrected within the documentation and codebase. (:pull:`1576`).
 
 Internal changes
 ^^^^^^^^^^^^^^^^
 * The `flake8` configuration has been migrated from `setup.cfg` to `.flake8`; `setup.cfg` has been removed. (:pull:`1569`)
 * The `bump-version.yml` workflow has been adjusted to bump the `patch` version when the last version is determined to have been a `release` version; otherwise, the `build` version is bumped. (:issue:`1557`, :pull:`1569`).
 * The GitHub Workflows now use the `step-security/harden-runner` action to monitor source code, actions, and dependency safety. All workflows now employ more constrained permissions rule sets to prevent security issues. (:pull:`1577`).
+* The `codespell` library is now a development dependency for the `dev` installation recipe with configurations found within `pyproject.toml`. This is also now a linting step and integrated as a `pre-commit` hook. For more information, see the `codespell documentation <https://github.com/codespell-project/codespell>`_ (:pull:`1576`).
```
Comment (coxipi) [inline CHANGES.rst:35]: Naive/uninformed question: Is the `dev` installation automatically used when git cloning xclim and doing `mamba install -n my_env -f environment.yml`. I think  `xscen` has  `environment_dev.yml`, but not `xclim`, so how does this work here?

**Review by coxipi — APPROVED** [review]
Nice to have. Seems to be working well, approved!

Note: There were some changes like x -> `x`. codespell does not make suggestions like this correct, just some unrelated completely manual changes while we're at it correct?

File: docs/notebooks/ensembles.ipynb (hunk 1 above)
Comment (Zeitsperre) [reply docs/notebooks/ensembles.ipynb:21]: Correct, haha. Bad habits die hard.

Comment (coxipi) [reply docs/notebooks/ensembles.ipynb:21]: Having this small interaction just to make sure is still more efficient than having a whole PR for such a small change IMO. Maybe such changes should be grouped, and maybe not the best practice, but there are certainly pros and cons

File: CHANGES.rst (hunk 2 above)
Comment (Zeitsperre) [reply CHANGES.rst:35]: That's how we've organized `xclim` here. When someone installs `xclim` into a `conda` environment with the `dev` recipe, it pulls the development libraries from PyPI. This hasn't been an issue to date, but if we wanted to mirror the `xscen` approach, adding this file into the codebase is feasible.

Perhaps worth discussing in a separate issue?

File: docs/notebooks/ensembles.ipynb (hunk 1 above)
Comment (Zeitsperre) [reply docs/notebooks/ensembles.ipynb:21]: As the code base is getting more and more cleaned up, I feel that this is becoming less and less of a concern, thankfully. But you're right that it would be good to have a comprehensive/in-depth plunge into the code to see if there are any more tiny bugs/issue worth cleaning up.

File: CHANGES.rst (hunk 2 above)
Comment (coxipi) [reply CHANGES.rst:35]: I don't see a problem with this organization, I just wanted to understand it, to be sure I'm not using a wrong version by error.
